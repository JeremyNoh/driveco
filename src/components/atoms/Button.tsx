import React from 'react';
import {
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  ViewStyle,
} from 'react-native';

interface Props {
  style?: ViewStyle;
  title: string;
  onPress: () => void;
  disabled?: boolean;
  withoutStyle?: boolean;
  styleText?: TextStyle;
}

const Button: React.FC<Props> = ({
  style,
  title,
  onPress,
  disabled = false,
  withoutStyle = false,
  styleText,
}) => {
  return (
    <TouchableOpacity
      style={[
        !withoutStyle && styles.button,
        style,
        disabled && styles.disabled,
      ]}
      onPress={onPress}
      disabled={disabled}>
      <Text style={[styles.text, styleText]}>{title}</Text>
    </TouchableOpacity>
  );
};

export default Button;

const styles = StyleSheet.create({
  button: {
    margin: 20,
    borderColor: '#136397',
    borderWidth: 1,
    alignItems: 'center',
    padding: 10,
    textAlign: 'center',
    justifyContent: 'center',
  },
  disabled: {
    pointerEvents: 'none',
    opacity: 0.5,
    backgroundColor: '#88B0CB',
  },
  text: {
    textTransform: 'uppercase',
    fontWeight: 'bold',
    margin: 0,
    textAlign: 'center',
  },
});
