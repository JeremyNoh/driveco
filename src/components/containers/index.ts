import TryLibContainer from './TryLibContainer';

export {TryLibContainer};
