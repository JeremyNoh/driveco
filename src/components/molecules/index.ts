import NfcTag from './NfcTag';
import ChangeLanguage from './ChangeLanguage';

export {NfcTag, ChangeLanguage};
