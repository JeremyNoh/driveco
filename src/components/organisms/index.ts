import TryLibOrganism from './TryLibOrganism';

export {TryLibOrganism};
