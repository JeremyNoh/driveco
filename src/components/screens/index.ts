import TryLibScreen from './TryLibScreen';

export {TryLibScreen};
