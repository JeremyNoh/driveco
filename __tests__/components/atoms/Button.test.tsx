import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import {cleanup, fireEvent, render} from '@testing-library/react-native';
import {Button} from '../../../src/components/atoms';

afterEach(cleanup);

describe('Button Atom', () => {
  it('renders correctly', () => {
    render(<Button onPress={() => {}} title="test" />).toJSON();
  });
  it('Match with snapshot', () => {
    const button = render(<Button onPress={() => {}} title="test" />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it('should show good Title', () => {
    const title = 'Test button';

    const {getByText} = render(<Button onPress={() => {}} title={title} />);

    const foundTitle = getByText(title);

    expect(foundTitle.props.children).toEqual(title);
  });
  it('should call function', () => {
    const title = 'Test button';
    const onPressMock = jest.fn();
    const eventData = {
      nativeEvent: {
        pageX: 20,
        pageY: 30,
      },
    };

    const {getByTestId} = render(
      <Button onPress={onPressMock} title={title} />,
    );

    fireEvent.press(getByTestId('buttonId'), eventData);

    expect(onPressMock).toHaveBeenCalledWith(eventData);
  });
});
