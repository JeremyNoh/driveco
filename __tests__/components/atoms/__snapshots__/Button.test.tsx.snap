// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Atom Match with snapshot 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#136397",
      "borderWidth": 1,
      "justifyContent": "center",
      "margin": 20,
      "opacity": 1,
      "padding": 10,
      "textAlign": "center",
    }
  }
  testID="buttonId"
>
  <Text
    style={
      Array [
        Object {
          "fontWeight": "bold",
          "margin": 0,
          "textAlign": "center",
          "textTransform": "uppercase",
        },
        undefined,
      ]
    }
    testID="textButtonId"
  >
    test
  </Text>
</View>
`;
